lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -41,14 +41,23 @@ def run(delay_sync: 3.seconds, reminder_interval: 30.seconds)
     def report_what_just_happened(resources, watch_time)
       new_successes, new_failures = resources.partition(&:deploy_succeeded?)
       new_failures.each do |resource|
+        tags = %W(namespace:#{@namespace} sha:#{ENV['REVISION']} resource:#{resource.id})
         if resource.deploy_failed?
+          ::StatsD.increment('deploy.resource.status', 1, tags: tags << "status:failed")
           @logger.error("#{resource.id} failed to deploy after #{watch_time}s")
         else
+          ::StatsD.increment('deploy.resource.status', 1, tags: tags << "status:timeout")
           @logger.error("#{resource.id} deployment timed out")
         end
       end
 
       if new_successes.present?
+        new_successes.each do |success|
+          tags = %W(namespace:#{@namespace} sha:#{ENV['REVISION']} resource:#{success.id})
+          ::StatsD.increment('deploy.resource.status', 1, tags: tags << "status:success")
+          ::StatsD.measure('deploy.resouce.duration', watch_time * 1000, tags: tags)
```

Comment:
Yes, but maybe its more inline with the reporting in the logs if we also report it on failures